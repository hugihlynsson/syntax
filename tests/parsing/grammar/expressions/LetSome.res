let foo = (x, y) => {
  let Some(vx) = x
  let x1 = vx + 1
  let Some({vy, vz: 3}) = y
  let y1 = vy + 1
  Some(x1 + y1)
}
