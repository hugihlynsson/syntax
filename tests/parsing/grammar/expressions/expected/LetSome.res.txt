let foo x y =
  ((match x with
    | Some vx ->
        let x1 = vx + 1 in
        (match y with
         | Some { vy; vz = 3 } -> let y1 = vy + 1 in Some (x1 + y1)
         | _ -> None)
    | _ -> None)
  [@ns.braces ])